mergify_engine/queue/merge_train.py

```diff
@@ -1205,9 +1219,16 @@ async def add_pull(
                 and config["effective_priority"]
                 > embarked_pull.config["effective_priority"]
             ):
-                # We found a car with lower priorit
+                # We found a car with lower priority
                 best_position = position
 
+        if need_to_be_readded:
+            # FIXME(sileht): this can be optimised by not dropping spec checks,
+            # if the position in the queue does not change
+            await self.remove_pull(ctxt)
+            await self.add_pull(ctxt, config)
```

Comment:
Is it worth having a `move_pull` which might be more Redis friendly (e.g. one operation or pipe)

Reply:
From Redis point of view, it would be 1  `hset()` instead 2. But this doesn't only move things in Redis, it may also clean up some resources on GitHub because some TrainCar has been deleted.